src/auth/src/error.rs

```diff
@@ -89,7 +89,7 @@ impl ErrorExt for Error {
             Error::FileWatch { .. } => StatusCode::InvalidArguments,
             Error::InternalState { .. } => StatusCode::Unexpected,
             Error::Io { .. } => StatusCode::StorageUnavailable,
-            Error::AuthBackend { .. } => StatusCode::Internal,
+            Error::AuthBackend { .. } => StatusCode::External,
```

Comment:
```suggestion
            Error::AuthBackend { source, .. } => source.status_code(),
```

Since `AuthBackend` is a `BoxedError`, we can use the actual error's source code instead.